### PR TITLE
Cleaning up TODOs

### PR DESCRIPTION
**Pull request by john-science**
## Description

ARMI has like 50 `TODO`s, which I want to remove and make tickets (or just fix).  But since we have a code freeze, that will have to wait.  

We also had like 3 `FIXME`s and a few `XXX`s.   But some of those were meant to be `TODO`s, and some were meant to be `NOTE`s.  Since even _I_ found it confusing, I am just replacing all `FIXME` and `XXX` with `NOTE` or `TODO`, and putting it in the docs that we don't allow `FIXME` or `XXX` any more.

I am just aiming to make the project more clear.

---

## Checklist

- [X] This PR has only one purpose or idea.
- [X] Tests have been added/updated to verify that the new/changed code works.

- [X] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [X] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [X] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any important changes.
- [X] The documentation is still up-to-date in the `doc` folder.
- [X] The dependencies are still up-to-date in `setup.py`.

### CODE REVIEW

File: armi/utils/tests/test_triangle.py [hunk 1]
```diff
@@ -88,7 +87,9 @@ def test_checkIfPointIsInTriangle(self):
         y3 = 0.376
         xP = 0.0
         yP = 0.17
-        # FIXME: generalTriangleInOrOut = triangle.checkIfPointIsInTriangle(xT1, yT1, xT2, yT2, xT3, yT3, xP, yP)
+        generalTriangleInOrOut = triangle.checkIfPointIsInTriangle(
+            xT1, yT1, xT2, yT2, xT3, yT3, xP, yP
+        )
```
Comment (opotowsky) [inline armi/utils/tests/test_triangle.py:92]: This cracks me up

Comment (opotowsky) [reply armi/utils/tests/test_triangle.py:92]: hashtag fixed

**Review by opotowsky — APPROVED** [review]
Easy little fixxer upper!

File: armi/utils/tests/test_triangle.py (hunk 1 above)
Comment (john-science) [reply armi/utils/tests/test_triangle.py:92]: lol

Yeah, my guess is the concerning bug was fixed years ago, probably right after this test line was commented out.

**Comment by alexhjames** [comment]
@john-science @opotowsky Maybe I missed it in the communications but this type of change would be nice to socialize a bit more. Oxbow also points to this pylintrc file because we stylistically try to maintain similarity to ARMI and also the whole pylint license issue. This change caused all of the FIXMEs in Oxbow to disappear in pylint and took a little bit to track down (I thought it was my pylint setup at first, not a PR change)

**Comment by opotowsky** [comment]
Oh I had no idea this could have any downstream impact!!

There is [an upcoming PR](https://github.com/terrapower/armi/pull/1296) that will remove `pylint` in favor of `ruff`. Would you like us to hold off? Can you copy the ARMI pylintrc file (minus this change) for your use?

**Pull request merged**